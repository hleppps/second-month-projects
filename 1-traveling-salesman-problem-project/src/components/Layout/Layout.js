import React from "react";
import Header from "../Header";
import styles from "./Layout.module.css"

const Layout = React.memo((props) => {
  return (
    <div className={styles.layout}>
      <Header changeThemeHandler={props.changeThemeHandler} />
      <main className={styles.main}>{props.children}</main>
    </div>
  );
});

export default Layout;
