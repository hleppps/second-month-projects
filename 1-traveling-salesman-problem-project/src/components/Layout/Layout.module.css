.layout {
  height: calc(100vh - 5rem);;
}

.main {
  margin-top: 5rem;
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 100%;
}