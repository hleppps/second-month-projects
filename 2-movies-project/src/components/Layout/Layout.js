import Header from "../Header";
import { Container } from "react-bootstrap";
import styles from "./Layout.module.scss";

const Layout = (props) => {
  return (
    <div className={styles.layout}>
      <Header onShowModal={props.onShowModal} isLoggedIn={props.isLoggedIn} setIsLoggedIn={props.setIsLoggedIn} />
      <main>
        <Container>{props.children}</Container>
      </main>
    </div>
  );
};

export default Layout;
