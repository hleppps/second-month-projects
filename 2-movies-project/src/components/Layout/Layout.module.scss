.layout main {
  padding: 20px 0;
}