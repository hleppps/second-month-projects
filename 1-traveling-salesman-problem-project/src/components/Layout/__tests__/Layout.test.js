import "@testing-library/jest-dom";
import { cleanup, render } from "@testing-library/react";
import React from "react";
import Layout from "../Layout";

afterEach(cleanup);

it("matches a snapshot", () => {
  const { asFragment } = render(<Layout />);
  expect(asFragment(<Layout />)).toMatchSnapshot();
});