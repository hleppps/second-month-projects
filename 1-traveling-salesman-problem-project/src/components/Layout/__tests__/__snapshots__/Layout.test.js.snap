// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header
      class="header"
    >
      <h1>
        Header
      </h1>
      <button
        class="logo"
        data-testid="button"
      />
    </header>
    <main
      class="main"
    />
  </div>
</DocumentFragment>
`;
